### PR TITLE
Add crypto benchmarks for iroha-ed25519 and Ursa

### PR DESCRIPTION
**Pull request by lebdron**
### Description of the Change
Keypair generation, signing, and verification benchmarks for iroha-ed25519 and Ursa

### Benefits
Can compare the performance of two libraries given our configuration

### Possible Drawbacks 
None

### Usage Examples or Tests *[optional]*
bm_iroha_ed25519, bm_ursa_ed25519

### CODE REVIEW

File: test/benchmark/bm_iroha_ed25519.cpp [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+#include <ed25519/ed25519.h>
+
+#include <cstdlib>
+#include <vector>
+
+#include <benchmark/benchmark.h>
+
+auto ConstructRandomVector(int64_t size) {
+  std::vector<unsigned char> v;
+  v.reserve(static_cast<int>(size));
```
Comment (igor-egorov) [inline test/benchmark/bm_iroha_ed25519.cpp]: The cast of int64_t to int looks very suspicious.

File: test/benchmark/bm_iroha_ed25519.cpp [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  for (int i = 0; i < size; ++i) {
+    v.push_back(static_cast<unsigned char>(std::rand() % size));
+  }
+  return v;
+}
+
+static void BM_CreateKeypair(benchmark::State &state) {
+  public_key_t pub{};
```
Comment (igor-egorov) [inline test/benchmark/bm_iroha_ed25519.cpp]: Looks like you do not need {} here and below.

File: test/benchmark/bm_iroha_ed25519.cpp [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+  while (state.KeepRunning()) {
+    state.PauseTiming();
+    auto data = ConstructRandomVector(state.range(0));
+    state.ResumeTiming();
+
+    ed25519_sign(&sig, data.data(), data.size(), &pub, &priv);
+  }
+}
+BENCHMARK(BM_Sign)->RangeMultiplier(2)->Range(1 << 10, 1 << 18);
```
Comment (igor-egorov) [inline test/benchmark/bm_iroha_ed25519.cpp]: 🤞 👉 👷

File: test/benchmark/bm_ursa_ed25519.cpp [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+#include <cstdlib>
+#include <vector>
+
+#include <benchmark/benchmark.h>
+
+auto ConstructRandomVector(int64_t size) {
+  std::vector<uint8_t> v;
+  v.reserve(static_cast<int>(size));
+  for (int i = 0; i < size; ++i) {
+    v.push_back(static_cast<uint8_t>(std::rand() % size));
```
Comment (igor-egorov) [inline test/benchmark/bm_ursa_ed25519.cpp]: on line 17 in bm_iroha_ed...cpp you have been using "unsigned char." Probably it worth to use the same type alias in both cases for consistency.

File: test/benchmark/bm_iroha_ed25519.cpp (hunk 2 above)
Comment (lebdron) [reply test/benchmark/bm_iroha_ed25519.cpp]: Unfortunately they are needed to initialize the struct which does not have a constructor.

File: test/benchmark/bm_iroha_ed25519.cpp [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+#include <cstdlib>
+#include <vector>
+
+#include <benchmark/benchmark.h>
+
+auto ConstructRandomVector(int64_t size) {
+  std::vector<unsigned char> v;
+  v.reserve(static_cast<int>(size));
+  for (int i = 0; i < size; ++i) {
+    v.push_back(static_cast<unsigned char>(std::rand() % size));
```
Comment (luckychess) [inline test/benchmark/bm_iroha_ed25519.cpp]: Shouldn't we call `srand` before somewhere?

**Comment by luckychess** [comment]
Oh, looks like CI should be fixed. A missing dependency?

**Comment by lebdron** [comment]
@luckychess A conditional is required for Ursa target, will add it

File: test/benchmark/bm_iroha_ed25519.cpp (hunk 5 above)
Comment (lebdron) [reply test/benchmark/bm_iroha_ed25519.cpp]: https://en.cppreference.com/w/cpp/numeric/random/rand

>If rand() is used before any calls to srand(), rand() behaves as if it was seeded with srand(1). 

This behavior should be fine here.

File: test/benchmark/bm_iroha_ed25519.cpp (hunk 1 above)
Comment (lebdron) [reply test/benchmark/bm_iroha_ed25519.cpp]: It should indeed work without it, thanks!

**Pull request merged**